actors/builtin/market/market_actor.go

```diff
@@ -462,6 +462,12 @@ func (a Actor) CronTick(rt Runtime, params *adt.EmptyValue) *adt.EmptyValue {
 				state, found, err := msm.dealStates.Get(dealID)
 				builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to get deal state")
 
+				// if this is the first cron tick for the deal, it should be in the pending state.
+				if state.LastUpdatedEpoch == epochUndefined {
+					pdErr := msm.pendingDeals.Delete(adt.CidKey(dcid))
+					builtin.RequireNoErr(rt, pdErr, exitcode.ErrIllegalState, "failed to delete pending proposal")
+				}
+
```

Comment:
I don't think this works. If this is the first cron tick for the deal, `state.LastUpdatedEpoch` will be 0 because it isn't set to `epochUndefined` until activation.

This should probably be included in the next block, `if !found { ... }`

Reply:
@wadeAlexC I've included a fix to remove this deal from pending even if it's timed out.